test/module/irohad/synchronizer/synchronizer_test.cpp

```diff
@@ -302,19 +378,26 @@ TEST_F(SynchronizerTest, ExactlyThreeRetrievals) {
 TEST_F(SynchronizerTest, RetrieveBlockTwoFailures) {
   DefaultValue<expected::Result<std::unique_ptr<MutableStorage>, std::string>>::
       SetFactory(&createMockMutableStorage);
-  EXPECT_CALL(*mutable_factory, createMutableStorage()).Times(1);
+  EXPECT_CALL(*mutable_factory, createMutableStorage()).Times(3);
```

Comment:
As I remeber, this test was made for `while(true)` case, so with the new change
https://github.com/hyperledger/iroha/blob/9618614d78c5d13c39786c1002082a6d7b0770c7/test/module/irohad/synchronizer/synchronizer_test.cpp#L62
it should try to retrieve the chain at least 11 times in this test.